### PR TITLE
Java: Update benchmarking to use CachedThreadPool 

### PR DESCRIPTION
**Pull request by acarbonetto**
See benchmarking results in: https://github.com/aws/glide-for-redis/issues/691#issuecomment-1980111310

replaces: https://github.com/aws/glide-for-redis/issues/691

### CODE REVIEW

File: java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java [hunk 1]
```diff
@@ -24,6 +29,8 @@ public class Benchmarking {
     static final int SIZE_SET_KEYSPACE = 3000000;
     public static final double NANO_TO_SECONDS = 1e9;
 
+    static final int NUM_OF_THREADS_TO_EXECUTE = 12;
```
Comment (Yury-Fridlyand) [inline java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java:32]: Do you need to bind this to the num of CPUs? What if it is executed on a VN with 1 CPU or with 128 CPUs?

File: java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java [hunk 2]
```diff
@@ -107,6 +114,23 @@ public static void printResults(
...
+                            Integer.MAX_VALUE,
+                            60L,
+                            TimeUnit.SECONDS,
+                            new SynchronousQueue<Runnable>(),
+                            (r, poolExecutor) -> {
+                                if (!poolExecutor.isShutdown()) {
+                                    try {
+                                        poolExecutor.getQueue().put(r);
+                                    } catch (InterruptedException e) {
+                                        throw new RuntimeException("interrupted");
```
Comment (Yury-Fridlyand) [inline java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java:130]: restore interrupt is missing

File: java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java [hunk 3]
```diff
@@ -107,6 +114,23 @@ public static void printResults(
     public static void testClientSetGet(
             Supplier<Client> clientCreator, BenchmarkingApp.RunConfiguration config, boolean async) {
         for (int concurrentNum : config.concurrentTasks) {
+            // same as Executors.newCachedThreadPool() with a RejectedExecutionHandler for robustness
+            ExecutorService executor =
+                    new ThreadPoolExecutor(
+                            0,
+                            Integer.MAX_VALUE,
```
Comment (Yury-Fridlyand) [inline java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java:121]: Shouldn't we keep some soft limits, like 4xN CPUS?

File: java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java (hunk 1 above)
Comment (acarbonetto) [reply java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java:32]: This is not used.  Removing.

File: java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java (hunk 3 above)
Comment (acarbonetto) [reply java/benchmarks/src/main/java/glide/benchmarks/utils/Benchmarking.java:121]: This is what the defaults are using.  Should we limit this?

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**